### DISCUSSION TITLE
What is the solution if a migration history conflict happens in production?

### DISCUSSION THREAD

**Discussion by cryptrr** [prisma/prisma#22439] [Q&A]
### Question

I know that for drift, you can use npx prisma migrate diff and resolve.

But was unable to find an answer for migration conflicts in production. Which can happen easily with botched git operations or if someone hotfixes directly to the database or if data loss happens somehow.

1) baselining to the current database state

2) Creating empty migration folders for each missing migration

3) npx prisma migrate resolve --applied migration for each missing migration

Would this work?

Or is it just possible to delete all the prisma migration entries from the database since baselining already synced the db with local.

There should be some way to fix this.

**Comment by ludralph** [comment]
Hi @cryptrr 👋 

Thank you for raising this question.

Baselining a database which you've mentioned will work and to resolve a migration history conflict in production with Prisma, you can follow these steps:

1. Replicate the change made in production in your Prisma schema. For example, if an index was added manually in the production database, you would add an @@index to the corresponding model in your schema.

2. Generate a new migration with the changes and take note of the full migration name, including the timestamp. You can do this by running the following command:

```
npx prisma migrate dev --name retroactively-add-index
```
This will create a new migration file in your migrations directory.

3. Push the migration to production without running migrate deploy. Instead, mark the migration as 'already applied' using the prisma migrate resolve command. This will add the migration to the migration history table without running the actual SQL. Here's the command to mark the migration as applied:
```
prisma migrate resolve --applied "20210316150542-retroactively-add-index"
```
4. If you have other databases that were patched, such as a staging database, repeat the previous step for each one.

5. For databases that were not patched, you can propagate the migration by committing it to source control and allowing your CI/CD pipeline to apply it to all databases. The migration will not be applied to databases where it has been marked as already applied.

You can see more from the [documentation](https://www.prisma.io/docs/orm/prisma-migrate/workflows/patching-and-hotfixing#migration-history-conflicts).

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by cryptrr** [reply]
The problem was that I lost already applied migration files. 

I already did

`prisma migrate resolve --applied ` after baselining.

Still it shows this when running 

`
prisma migrate dev --create-only
` 

`
The following migration(s) are applied to the database but missing from the local migrations directory 
`

Is there a way to simply reset the migrations history in the database itself to start from zero.

**Reply by cryptrr** [reply]
Does deleting the migrations history table from the database and starting over again work?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
